source/server/http/admin.cc

```diff
@@ -284,17 +284,18 @@ void AdminImpl::writeClustersAsJson(Buffer::Instance& response) {
         envoy::admin::v2alpha::HostStatus& host_status = *cluster_status.add_host_statuses();
         Network::Utility::addressToProtobufAddress(*host->address(),
                                                    *host_status.mutable_address());
-
         for (const Stats::CounterSharedPtr& counter : host->counters()) {
-          auto& metric = (*host_status.mutable_stats())[counter->name()];
-          metric.set_type(envoy::admin::v2alpha::SimpleMetric::COUNTER);
+          auto& metric = *host_status.add_stats();
```

Comment:
IIUC, the order that you iterate through these stats is dictated by their order in this [unordered_map](https://github.com/envoyproxy/envoy/blob/c68063c05d940cfee1803f13bf2612bd56d01035/source/common/stats/isolated_store_impl.h#L51).  In what situations would you like to have the ordering stay the same?  I think your test could break in the future if the unordered_map internals (like its hash function) change.  There are a few solutions here:
1.  Change the `std::unordered_map` in the isolated store to an `std::map` everywhere or by specifying the data structure as a template parameter so it could be done in the Host object case only.
2.  Sort the entries when exporting.